### DISCUSSION TITLE
Particles to mesh

### DISCUSSION THREAD

**Discussion by SimonNgj** [AcademySoftwareFoundation/openvdb#1402] [General]
I have a point cloud and I would like to do mesh reconstruction for that point cloud. At the moment, I am following this procedure:
1. Consider point cloud as particles
2. Convert particles to Level set
3. Apply Dilation -> Smooth filter -> erosion -> convert volumeToMesh
I get the mesh after applying these steps, but the the mesh is in a rim shape. I want the mesh to be a single layer, is there any way to achieve that?
I attach below an image to illustrate want to achieve
![Capture](https://user-images.githubusercontent.com/22513384/170937618-bef1f0e2-2972-4d1e-98a9-4613a64fe802.PNG)
.

**Comment by SimonNgj** [comment]
From this image, (a) is what I have. You can see that there is an inner mesh inside the outer mesh. (b) is what I want to achieve
![Untitled](https://user-images.githubusercontent.com/22513384/170939146-4d9cc3b4-2388-4cb1-ac7f-aeb5705183c2.png)
.

**Comment by Idclip** [comment]
Hello @SimonNgj - as with your other issue, `LevelSetRebuild.h` or the tools in `FastSweeping.h` should be a good start

**Comment by apradhana** [comment]
@SimonNgj Another method I've had success with is by adding more points inside of your geometry. Once I've converted the _original set of points_ to a level-set, then I use one of the methods in `PointScatter.h` to add more points inside of the level-set. I then make another `particlesToSdf` call before applying the level-set operations (i.e. dilation, smoothing, and erosion) you mentioned.
